core/src/main/scala/org/apache/spark/sql/delta/GeneratedColumn.scala

```diff
@@ -362,6 +362,31 @@ object GeneratedColumn extends DeltaLogging with AnalysisHelper {
                   createExpr(name)(SubstringPartitionExpr(partColName, pos, len))
                 case ExtractBaseColumn(name, _) =>
                   createExpr(name)(IdentityPartitionExpr(partColName))
+                case TruncDate(ExtractBaseColumn(name, DateType), StringLiteral(format)) =>
```

Comment:
I implemented it that way to validate the format was correct. Should I add the validation inside TruncDatePartitionExpr and remove it from here?

Reply:
I tested two scenarios one with validation and another without validation using incorrect format:

In the first case it will not add the partition filter to the plan, in the second case it adds the partition filter to the plan. In both cases the column will always be null because of the incorrect format, so the behavior of the query will not get affected. I will pass the format directly.